mail_tracking/static/src/js/failed_message.js

```diff
@@ -0,0 +1,384 @@
+        }),
+
+        _sidebarQWebParams: function () {
+            var failed = this.call('mail_service', 'getMailbox', 'failed');
+            return {
+                activeThreadID: this._thread ? this._thread.getID() : undefined,
+                failedCounter: failed.getMailboxCounter(),
+            };
+        },
+
+        _renderSidebar: function () {
+            var $sidebar = this._super.apply(this, arguments);
+            // Because Odoo implementation isn't designed to be inherited
+            // properly we inject failed button using jQuery.
+            var $sidebarFailed = $(QWeb.render('mail_tracking.SidebarFailed',
+                this._sidebarQWebParams()));
+            $sidebarFailed.insertBefore($sidebar.find("hr[class='mb8']"));
+            return $sidebar;
+        },
+
+        _onMessageUpdated: function (message, type) {
+            var self = this;
+            var currentThreadID = this._thread.getID();
+            if (currentThreadID === 'mailbox_failed' && !message.isFailed()) {
+                this._thread.fetchMessages(this.domain)
+                    .then(function () {
+                        var options = self._getThreadRenderingOptions();
+                        self._threadWidget.removeMessageAndRender(
+                            message.getID(), self._thread, options)
+                            .then(function () {
+                                self._updateButtonStatus(
+                                    !self._thread.hasMessages(), type);
+                            });
+                    });
+            } else {
+                this._super.apply(this, arguments);
+            }
+        },
+
+        _getThreadRenderingOptions: function () {
+            var values = this._super.apply(this, arguments);
+            if (this._thread.getID() === 'mailbox_failed') {
+                values.displayEmailIcons = true;
+                values.displayReplyIcons = false;
+                values.displayRetryButton = true;
+                values.displayReviewedButton = true;
+            }
+            return values;
+        },
+
+        _startListening: function () {
+            this._super.apply(this, arguments);
+            this.call('mail_service', 'getMailBus')
+                .on('update_failed', this, this._throttledUpdateThreads);
+        },
+
+        // Handlers
+        _onRetryFailedMessage: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this.do_action('mail.mail_resend_message_action', {
+                additional_context: {
+                    mail_message_to_resend: messageID,
+                },
+            });
+        },
+
+        _onMarkFailedMessageReviewed: function (event) {
+            event.preventDefault();
+            var messageID = $(event.currentTarget).data('message-id');
+            this._rpc({
```

Comment:
Think of you code as extendable by default.

```suggestion
            return this._rpc({
```